Sandbox/Main.cs

```diff
@@ -12,6 +17,111 @@ public partial class Main : Form
         public Main()
         {
             InitializeComponent();
+            var plotInput = chMain.Series.Add("Input");
+            plotInput.ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.FastPoint;
+            var plotOutput = chMain.Series.Add("Output");
+            plotOutput.ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.FastLine;
+            chMain.ChartAreas[0].AxisX.IsLogarithmic = true;
+            // Build circuit
+            Circuit ckt = new Circuit();
+            ckt.Objects.Add(
+                new VoltageSource("V1", "in", "0", 0.0),
+                new VoltageSource("Vsupply", "vdd", "0", 5.0),
+                new Resistor("R1", "vdd", "out", 1.0e3),
+                new Resistor("R2", "out", "b", 10.0e3),
+                new Capacitor("Cin", "in", "b", 1e-6),
+                CreateBJT("Q1", "c", "b", "0", "0", "mjd44h11", string.Join(" ",
+                    "IS = 1.45468e-14 BF = 135.617 NF = 0.85 VAF = 10",
+                    "IKF = 5.15565 ISE = 2.02483e-13 NE = 3.99964 BR = 13.5617",
+                    "NR = 0.847424 VAR = 100 IKR = 8.44427 ISC = 1.86663e-13",
+                    "NC = 1.00046 RB = 1.35729 IRB = 0.1 RBM = 0.1",
+                    "RE = 0.0001 RC = 0.037687 XTB = 0.90331 XTI = 1",
+                    "EG = 1.20459 CJE = 3.02297e-09 VJE = 0.649408 MJE = 0.351062",
+                    "TF = 2.93022e-09 XTF = 1.5 VTF = 1.00001 ITF = 0.999997",
+                    "CJC = 3.0004e-10 VJC = 0.600008 MJC = 0.409966 XCJC = 0.8",
+                    "FC = 0.533878 CJS = 0 VJS = 0.75 MJS = 0.5",
+                    "TR = 2.73328e-08 PTF = 0 KF = 0 AF = 1"))
+                );
+            ckt.Objects["V1"].ParameterSets.SetProperty("acmag", 1.0);
+
+            // Create simulation
+            AC ac = new AC("ac", new SpiceSharp.Simulations.Sweeps.DecadeSweep(10, 10e9, 5));
+
+            // Create exports
+            Func<ComplexState, Complex>[] exports = new Func<ComplexState, Complex>[1];
+            ac.InitializeSimulationExport += (object sender, InitializationDataEventArgs args) =>
+            {
+                exports[0] = ac.CreateACVoltageExport("out");
```

Comment:
That should be possible. Although performance will drop a little bit as the associated node index needs to be searched for every call.

Reply:
I will create a new export class to handle the validity of the export (exports should not be called before or after a simulation). I believe it will also be better to move the `Nodes` (currently in `Circuit`) to `Simulation`.